x/qgb/abci_test.go

```diff
@@ -526,3 +520,39 @@ func TestDataCommitmentCreationCatchup(t *testing.T) {
 		assert.Equal(t, want[i].Nonce, dc.Nonce)
 	}
 }
+
+// TestPruning tests the pruning mechanism via generating a set of attestations, then checking if the expired
+// ones are pruned
+func TestPruning(t *testing.T) {
+	input, ctx := testutil.SetupFiveValChain(t)
+	qgbKeeper := *input.QgbKeeper
+	// set the data commitment window
+	window := uint64(101)
+	qgbKeeper.SetParams(ctx, types.Params{DataCommitmentWindow: window})
+	initialBlockTime := ctx.BlockTime()
+	// make the interval between blocks being 3 days
+	ctx = testutil.ExecuteQGBHeightsWithTime(ctx, qgbKeeper, 1, 5000, time.Hour)
+
+	earliestAttestationNonce := qgbKeeper.GetEarliestAvailableAttestationNonce(ctx)
+	assert.LessOrEqual(t, earliestAttestationNonce, qgbKeeper.GetLatestAttestationNonce(ctx))
```

Comment:
what is this testing for? just that earliest and latest are working as expected?

Reply:
[302f9a2](https://github.com/celestiaorg/celestia-app/pull/1799/commits/302f9a26d4b097e9f5a5132871f8dcec4afe6e7b)